src/drivers/driver.ts

```diff
@@ -1235,23 +1235,32 @@ export abstract class CMakeDriver implements vscode.Disposable {
     const gen = this.generatorName;
     target = this.correctAllTargetName(target);
 
-    const generator_args = (() => {
-      if (!gen)
-        return [];
-      else if (/(Unix|MinGW) Makefiles|Ninja/.test(gen) && target !== 'clean')
-        return ['-j', this.config.numJobs.toString()];
-      else if (/Visual Studio/.test(gen) && target !== 'clean')
-        return ['/maxcpucount:' + this.config.numJobs.toString()];
-      else
-        return [];
-    })();
+    const buildArgs: string[] = this.config.buildArgs.slice(0);
+    const buildToolArgs: string[] = ['--'].concat(this.config.buildToolArgs);
+
+    // Prefer using CMake's build options to set parallelism over tool-specific switches.
+    // The feature is not available until version 3.14.
+    if (this.cmake.version && this.cmake.version.major >= 3 && this.cmake.version.minor > 13) {
+      buildArgs.push('-j');
+      if (this.config.numJobs) {
+        buildArgs.push(this.config.numJobs.toString());
+      }
+    } else {
+      if (gen) {
+        if (/(Unix|MinGW) Makefiles|Ninja/.test(gen) && target !== 'clean') {
+          buildToolArgs.push('-j', this.config.numJobs.toString());
+        } else if (/Visual Studio/.test(gen) && target !== 'clean') {
```

Comment:
Can this.config.numJobs be null here? I am asking because I noticed the check higher up when version >= 3.14.
Otherwise looks good.

Reply:
It's not checking for null.  It's checking for non-zero.

Reply:
Got it, I understand.